internal/services/cognitive/cognitive_account_rai_blocklist_resource.go

```diff
@@ -0,0 +1,216 @@
+package cognitive
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/cognitive/2024-10-01/raiblocklists"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type cognitiveRaiBlocklistModel struct {
+	Name               string `tfschema:"name"`
+	CognitiveAccountId string `tfschema:"cognitive_account_id"`
+	Description        string `tfschema:"description"`
+}
+
+type CognitiveRaiBlocklistResource struct{}
+
+var _ sdk.Resource = CognitiveRaiBlocklistResource{}
+
+func (c CognitiveRaiBlocklistResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"cognitive_account_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: raiblocklists.ValidateAccountID,
+		},
+
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+	}
+}
+
+func (c CognitiveRaiBlocklistResource) Attributes() map[string]*schema.Schema {
+	return map[string]*schema.Schema{}
+}
+
+func (c CognitiveRaiBlocklistResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model cognitiveRaiBlocklistModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.Cognitive.RaiBlocklistsClient
+			accountId, err := raiblocklists.ParseAccountID(model.CognitiveAccountId)
+			if err != nil {
+				return err
+			}
```

Comment:
This might seem minor but we review so many PRs that should by and large be boilerplate code. It reduces the cognitive load on the reviewer if the pattern that is prevalent through most of the provider is followed, so could we reorder this
```suggestion
			client := metadata.Client.Cognitive.RaiBlocklistsClient
			
			var model cognitiveRaiBlocklistModel
			if err := metadata.Decode(&model); err != nil {
				return fmt.Errorf("decoding: %+v", err)
			}

			accountId, err := raiblocklists.ParseAccountID(model.CognitiveAccountId)
			if err != nil {
				return err
			}
```